lib/src/main/java/org/esbtools/lightbluenotificationhook/NotificationHook.java

```
@@ -120,69 +120,61 @@ private HookResult processSingleHookDoc(EntityMetadata metadata,
      */
-    private boolean watchedFieldsHaveChanged(EntityMetadata metadata,
-                                             JsonDoc preDoc,
-                                             JsonDoc postDoc,
-                                             Projector watchProjector,
-                                             boolean arrayOrderingSignificant) {
+    private DocComparator.Difference<JsonNode> compareDocs(EntityMetadata metadata,
+                                                           JsonDoc preDoc,
+                                                           JsonDoc postDoc,
+                                                           Projector watchProjector) {
         JsonDoc watchedPostDoc = watchProjector.project(postDoc, jsonNodeFactory);
         JsonDoc watchedPreDoc = preDoc == null
             ? new JsonDoc(jsonNodeFactory.objectNode())
             : watchProjector.project(preDoc, jsonNodeFactory);
-
+        
         // Compute diff
         JsonCompare cmp=metadata.getDocComparator();
         try {
-            DocComparator.Difference diff=cmp.compareNodes(watchedPreDoc.getRoot(),watchedPostDoc.getRoot());
+            DocComparator.Difference<JsonNode> diff=cmp.
+                compareNodes(watchedPreDoc.getRoot(),watchedPostDoc.getRoot());
             LOGGER.debug("Diff: {}",diff);
-            if(!diff.same()) {
-                if(diff.getNumChangedFields()>0) {
-                    return true;
-                } else {
-                    // Are there any array element moves?
-                    // Do we care?
-                    if(!diff.getDelta().isEmpty()&&arrayOrderingSignificant) {
-                        return true;
-                    }
-                }
-            }
-            return false;
+            return diff;
         } catch (Exception e) {
             LOGGER.error("Error processing notification",e);
-            return false;
-        }
+            return null;
```

I think this can result in NPE. I say let exception propagate and force caller to deal with and return failed HookResult.